fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/LakeTableTieringManager.java

```diff
@@ -0,0 +1,252 @@
+    }
+
+    private void addLakeTable(TableInfo tableInfo, @Nullable Long delayMs) {
+        inLock(
+                lock,
+                () -> {
+                    tablePathById.put(tableInfo.getTableId(), tableInfo.getTablePath());
+                    tieringIntervalByTableId.put(
+                            tableInfo.getTableId(),
+                            tableInfo.getTableConfig().getDataLakeTieringInterval().toMillis());
+                    scheduleTableTiering(tableInfo.getTableId(), delayMs);
+                });
+    }
+
+    private void scheduleTableTiering(long tableId) {
+        scheduleTableTiering(tableId, null);
+    }
+
+    private void scheduleTableTiering(long tableId, @Nullable Long delayMs) {
+        Long tieringInterval = tieringIntervalByTableId.get(tableId);
+        if (tieringInterval == null) {
+            // the table has been dropped, return directly
+            return;
+        }
+        lakeTieringScheduleTimer.add(
+                new DelayedTiering(tableId, delayMs == null ? tieringInterval : delayMs));
+    }
+
+    public void removeLakeTable(long tableId) {
+        inLock(
+                lock,
+                () -> {
+                    tablePathById.remove(tableId);
+                    tieringIntervalByTableId.remove(tableId);
+                });
+    }
+
+    @Nullable
+    public Tuple2<Long, TablePath> requestTable() {
+        Long tableId = pendingTieringTables.poll();
+        // no any pending table, return directly
+        if (tableId == null) {
+            return null;
+        }
+        TablePath tablePath = tablePathById.get(tableId);
+        // the table has been dropped, request again
+        if (tablePath == null) {
+            return requestTable();
+        }
+        tieringLivenessProbe(tableId);
+        return new Tuple2<>(tableId, tablePath);
+    }
+
+    private void tieringLivenessProbe(long tableId) {
+        // we put the table to a timout timer,
+        // if no liveness for the table in this period of time, we consider the tiering is
+        // invalid and put it back to wait to be scheduled by other tiering service
+        lakeTieringScheduleTimer.add(
+                new TimerTask(TIERING_SERVICE_TIMEOUT_MS) {
+                    @Override
+                    public void run() {
+                        inLock(
+                                lock,
+                                () -> {
+                                    // the table is not in tiering, we put it back to wait to be
+                                    // scheduled
+                                    if (!liveTieringTableIds.remove(tableId)) {
+                                        scheduleTableTiering(tableId);
+                                    } else {
+                                        // schedule liveness probe again
+                                        tieringLivenessProbe(tableId);
+                                    }
```

Comment:
This logic is hard to understand. What's the purpose of this method? Detect no-heartbeat tiering table or stop tiering timeout table? 
- If detect no-heartbeat tiering table, it would be better to add `lastHeartbeatTime` in the `liveTieringTableIds` for each table. So it's easy to judge for the timeout timer. 
- If stop tiering timeout table, I think we don't need to "schedule liveness probe again", just send STOP event to the flink job.

Reply:
It's for detect no-heartbeat tiering table. It's good suggestion to add  `lastHeartbeatTime` to make it more clear.